samples/ar-view-node/src/main/java/io/github/sceneview/sample/arviewnode/MainActivity.kt

```diff
@@ -0,0 +1,162 @@
+package io.github.sceneview.sample.arviewnode
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.google.ar.core.Config
+import io.github.sceneview.ar.ARSceneView
+import io.github.sceneview.ar.arcore.createAnchorOrNull
+import io.github.sceneview.ar.node.AnchorNode
+import io.github.sceneview.sample.arviewnode.nodes.events.ImageNodeEvent
+import io.github.sceneview.sample.arviewnode.nodes.events.ViewNodeEvent
+import io.github.sceneview.sample.arviewnode.nodes.ImageNodeHelper
+import io.github.sceneview.sample.arviewnode.nodes.TrackingNode
+import io.github.sceneview.sample.arviewnode.nodes.ViewNodeHelper
+import io.github.sceneview.sample.arviewnode.utils.NodeRotationHelper
+
+
+class MainActivity : AppCompatActivity(), MainActivityViewMvc.Listener {
+
+    private lateinit var viewMvc: MainActivityViewMvc
+
+    private lateinit var trackingNode: TrackingNode
+    private lateinit var imageNodeHelper: ImageNodeHelper
+    private lateinit var viewNodeHelper: ViewNodeHelper
+    private lateinit var nodeRotationHelper: NodeRotationHelper
+
+    private val nodeList: MutableList<AnchorNode> = mutableListOf()
+
+    private var nodeType = NodeType.VIEW_NODE
+
+    ////////////////////
+    // region Lifecycle Methods
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        viewMvc = MainActivityViewMvcImpl(
+            this,
+            layoutInflater,
+            null
+        )
+        setContentView(viewMvc.getView().root)
+        viewMvc.registerListener(this)
+
+        viewMvc.getSceneView().apply {
+            initObjects(this)
+            configureArSession(this)
+            handleOnSessionUpdated(this)
+
+            lifecycle = this@MainActivity.lifecycle
+        }
```

Comment:
I'd rather use `let` or extension methods here.

Reply:
To avoid the call "this@MainActivity.lifecycle"?

Reply:
Yes, and also pass `it` instead of `this` as an argument to methods.

Reply:
fixed